tests/cupy_tests/core_tests/test_include.py

```diff
@@ -0,0 +1,79 @@
+import os
+from unittest import mock
+
+import cupy
+
+import pytest
+
+
+_code_base = '''
+#include <cuda_fp16.h>
+
+#include <cupy/carray.cuh>
+#include <cupy/complex.cuh>
+#include <cupy/cuComplex_bridge.h>
+#include <cupy/math_constants.h>
+#include <cupy/atomics.cuh>
+#include <cupy/hip_workaround.cuh>
+#include <cupy/swap.cuh>
+#include <cupy/tuple.cuh>
```

Comment:
Can we use Python to loop over the folder and pick up all headers programmatically? This way we would not miss any future additions.

Reply:
Yeah I tried it first but there is a dependency between headers 😞
(`atomics.cuh` requires `carray.cuh`)

Reply:
I see. It might be a good idea to defer this when re-considering #3612.

Reply:
(The context is one of the `atomicAdd` involves `float16`.)